aarch64/root/firmware-update.sh

```diff
@@ -45,6 +50,31 @@ if [[ -z "${fileName// }" ]]; then
         echo "Exiting..."
         exit 1
 fi
+filePath="${FW_PATH%/}/$fileName"
+if [[ ! -f $filePath ]]; then
+        echo "File does not exist. Trying download."
+        read -p "Enter Y to proceed, any other entry to exit: " answer
+        echo " "
+        if [[ $answer != [yY] ]]; then
+                echo "Exiting..."
+                exit 1
+        fi
+        echo "Downloading..."
+        echo " "
+        curl -o "$filePath" "${FW_BASE%/}/$fileName"
+        retVal=$?
+        if (( retVal != 0 )); then
+                echo " "
+                echo "Download Error! Please re-run this script..."
+                echo " "
+                exit $retVal
+        fi
+        echo " "
+        echo "Download complete! Checking md5 checksum:"
+        md5=$(curl -s "${FW_BASE%/}/${fileName}.md5")
+        echo "${md5% *} ${filePath}" | md5sum --check
+        echo " "
```

Comment:
The script should check the return value of md5sum and exit if there is a sum mismatch with a printed error message.